src/main/takeDetection/takeDetection.ts

```diff
@@ -48,56 +48,202 @@ export function findSentences(words: Word[]): Sentence[] {
+        sentences[nextSentenceIdxToCompare].sentenceString,
+        sentences[potentialTakeStartIdxs[j] + i].sentenceString
+      );
+
+      // if sentence in the chunk is not similar, discard all chunks to the right
+      if (nextSimilarity < THRESHOLD) {
+        const potentialTakeStartIdxsLen = potentialTakeStartIdxs.length;
+        const chunksToDiscard =
+          potentialTakeStartIdxsLen - potentialTakeStartIdxs[j];
+        const chunksLeft = potentialTakeStartIdxsLen - chunksToDiscard;
+
+        // if only the first chunk left, reset
+        if (chunksLeft === 1) {
+          // if potential take greater then two, re-start from the first sentence in the second chunk
+          if (potentialTakeStartIdxsLen > 2) {
+            updatedCurrentSentenceIdx = potentialTakeStartIdxs[j] + i - 1;
+          } else {
+            // else start from the failed sentence
+            updatedCurrentSentenceIdx = potentialTakeStartIdxs[j] + i;
+          }
+
+          isSimilarTake = false;
+          break;
+        }
+
+        // discard chunks to the right
+        for (let k = 0; k < chunksToDiscard; k += 1) {
+          potentialTakeStartIdxs.pop();
+        }
+      }
+    }
+  }
+
+  if (isSimilarTake)
+    updatedCurrentSentenceIdx =
+      potentialTakeLen * potentialTakeStartIdxs.length;
+
+  return { isSimilarTake, updatedCurrentSentenceIdx };
+};
+
+const newTakeGroup = (
+  potentialTakeStartIdxs: number[],
+  potentialTakeLen: number,
+  sentences: Sentence[]
+): InjectableTakeGroup => {
+  // if all the sentence in potential take are similar, add them to takeGroups
   const newGroup: InjectableTakeGroup = { takes: [] };
-  let nextTakeRange: IndexRange = {
-    startIndex: sentenceOne.startIndex as number,
-    endIndex: sentenceOne.endIndex as number,
-  };
-  let nextTake: InjectableTake = { wordRange: nextTakeRange };
-  newGroup.takes.push(nextTake);
-  nextTakeRange = {
-    startIndex: sentenceTwo.startIndex as number,
-    endIndex: sentenceTwo.endIndex as number,
-  };
-  nextTake = { wordRange: nextTakeRange };
-  newGroup.takes.push(nextTake);
+
+  potentialTakeStartIdxs.forEach((startIdx) => {
+    const takeStartSentence = sentences[startIdx];
+    const takeEndSentence = sentences[startIdx + potentialTakeLen - 1];
+
+    const takeRange: IndexRange = {
+      startIndex: takeStartSentence.startIndex as number,
+      endIndex: takeEndSentence.endIndex as number,
+    };
+    const take: InjectableTake = { wordRange: takeRange };
+    newGroup.takes.push(take);
+  });
+
   return newGroup;
-}
+};
 
-export function findTakes(words: Word[], threshold = THRESHOLD): InjectableTakeGroup[] {
-  const takeGroups: InjectableTakeGroup[] = [];
-  const sentences: Sentence[] = findSentences(words);
-  for (let i = 0; i < sentences.length - 1; i += 1) {
-    const similarity: number = getSimilarityScore(
-      sentences[i].sentenceString,
-      sentences[i + 1].sentenceString
+const startDetection: (
```

Comment:
Add a comment explaining what this function does, what the inputs and outputs are, any mutations etc